lib/shelley/src/Cardano/Wallet/Shelley/Api/Server.hs

```diff
@@ -285,10 +307,12 @@ server byron icarus shelley spl ntp =
                 (icarus, listAddresses icarus (const pure) wid s)
              )
 
-    byronCoinSelections :: Server (CoinSelections n)
-    byronCoinSelections wid x = withLegacyLayer wid
-        (byron, liftHandler $ throwE ErrNotASequentialWallet)
-        (icarus, selectCoins icarus (const $ paymentAddress @n) wid x)
+    byronCoinSelections :: Server (ByronCoinSelections n)
+    byronCoinSelections wid (ApiSelectForPayment x) =
+        withLegacyLayer wid
+            (byron, liftHandler $ throwE ErrNotASequentialWallet)
+            (icarus, selectCoins icarus (const $ paymentAddress @n) wid x)
+    byronCoinSelections _ _ = throwError err400
```

Comment:
What about having a nicer error message here? Something that tells clients that there's no such thing as selecting coins for delegation on a byron wallet because such wallets have no delegation capabilities whatsoever.